src/main/java/com/hubspot/smtp/client/SmtpSessionFactory.java

```diff
@@ -52,6 +66,11 @@ public SmtpSessionFactory(ExecutorService executorService) {
     return connectFuture;
   }
 
+  @Override
+  public void close() throws IOException {
+    allChannels.close().awaitUninterruptibly();
```

Comment:
I would make it interruptable and catch and rethrow (while calling `Thread.currentThread().interrupt()`). Otherwise if a jvm is really trying to shut down this might prevent it.